src/Tempest/Mailer/src/Components/Address/ImmutableAddressCollection.php

```diff
@@ -0,0 +1,10 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tempest\Mailer\Components\Address;
+
+class ImmutableAddressCollection
```

Comment:
Why do we need a mutable and immutable variant? Only immutable is fine I think

Reply:
Those addresses should be modifiable during the build phase but not after making the email object. We can't have just an immutable because then it wouldn't be modifiable at all.